### PR TITLE
Convert all of TriggerBuilds to a/a

### PR DESCRIPTION
**Pull request by finestructure**
I want to make changes for doc generation reporting and it's best to pull this forward before I do.

There are no functional changes in this PR, it's a pure conversion from ELF → a/a.

### CODE REVIEW

File: Sources/App/Commands/TriggerBuilds.swift [hunk 1]
```diff
@@ -184,68 +173,76 @@ func triggerBuilds(on database: Database,
...
-        return packages.map {
-            findMissingBuilds(database, packageId: $0).flatMap {
-                triggerBuildsUnchecked(on: database, client: client, logger: logger, triggers: $0) }
+        return await withThrowingTaskGroup(of: Void.self) { group in
+            for package in packages {
+                group.addTask {
+                    let triggerInfo = try await findMissingBuilds(database, packageId: package)
+                    try await triggerBuildsUnchecked(on: database, client: client, logger: logger, triggers: triggerInfo)
+                }
+            }
```
Comment (finestructure) [inline Sources/App/Commands/TriggerBuilds.swift:193]: TODO: double check if this is the correct conversion. We might be making the triggers concurrent when they're not in ELF-land.

File: Sources/App/Commands/TriggerBuilds.swift [hunk 2]
```diff
@@ -184,68 +173,76 @@ func triggerBuilds(on database: Database,
...
+    async let runningJobsTask = Current.getStatusCount(client, .running).get()
+    let pendingJobs = try await pendingJobsTask
+    let runningJobs = try await runningJobsTask
+
+    AppMetrics.buildPendingJobsCount?.set(pendingJobs)
+    AppMetrics.buildRunningJobsCount?.set(runningJobs)
+
+    let newJobs = ActorIsolated(0)
+
+    await withThrowingTaskGroup(of: Void.self) { group in
```
Comment (finestructure) [inline Sources/App/Commands/TriggerBuilds.swift:207]: Same here: double check concurrency behaviour. We don't want to be flooding Gitlab.

File: Sources/App/Commands/TriggerBuilds.swift [hunk 3]
```diff
@@ -256,47 +253,49 @@ func triggerBuilds(on database: Database,
...
-            return Build.trigger(database: database,
-                                 client: client,
-                                 logger: logger,
-                                 buildId: buildId,
-                                 platform: pair.platform,
-                                 swiftVersion: pair.swiftVersion,
-                                 versionId: trigger.versionId)
-                .flatMap { response in
+                            triggers: [BuildTriggerInfo]) async throws {
+    await withThrowingTaskGroup(of: Void.self) { group in
```
Comment (finestructure) [inline Sources/App/Commands/TriggerBuilds.swift:260]: And here: double check concurrency.

**Review by daveverwer — APPROVED** [review]

File: Sources/App/Commands/TriggerBuilds.swift (hunk 1 above)
Comment (finestructure) [reply Sources/App/Commands/TriggerBuilds.swift:193]: https://gist.github.com/finestructure/91f7d755c7d966751fd1a23e64b03013

**Comment by finestructure** [comment]
Merging to let it "sink in" on dev for a bit

**Comment by finestructure** [comment]
Working as expected:

CleanShot 2023-01-29 at 08 34 28@2x

CleanShot 2023-01-29 at 08 40 48@2x

CleanShot 2023-01-29 at 08 40 19@2x

**Pull request merged**